backend/app/utils/api_response.py

```diff
@@ -0,0 +1,47 @@
+"""API response classes."""
+from abc import ABC, abstractmethod
+from typing import Any, Dict
+
+class ApiResponse(ABC):
+    """Abstract base class for API responses."""
+
+    @abstractmethod
+    def to_dict(self) -> Dict[str, Any]:
+        """Convert the response to a dictionary."""
+        pass
+
+class SuccessResponse(ApiResponse):
```

Comment:
instead of defining two classes for Success and Error responses it's much better to move error as an attribute(like message and data) and also rename SuccessResponse to just Response.